src/algorithms/signal_source/adapters/file_source_base.cc

```diff
@@ -249,17 +259,24 @@ void FileSourceBase::disconnect(gr::top_block_sptr top_block)
     // DUMP
     if (sink())
         {
-            top_block->disconnect(std::move(output), 0, sink(), 0);
+            top_block->disconnect(output, 0, sink(), 0);
```

Comment:
It turns out, all these disconnect methods work too hard. Rather than trying to unwind all the connections, you can just `top_block->disconnect(output)` and the block is removed from the graph.